extensions/guacamole-auth-cas/src/main/java/org/apache/guacamole/auth/cas/AuthenticationProviderService.java

```diff
@@ -83,7 +115,15 @@ public AuthenticatedUser authenticateUser(Credentials credentials)
             String ticket = request.getParameter(CASTicketField.PARAMETER_NAME);
             if (ticket != null) {
                 AuthenticatedUser authenticatedUser = authenticatedUserProvider.get();
-                authenticatedUser.init(ticketService.processUsername(ticket), credentials);
+                AttributePrincipal principal = ticketService.validateTicket(ticket);
+                String username = principal.getName();
+                Object credObj = principal.getAttributes().get("credential");
+                if (credObj != null) {
```

Comment:
The complexity of what's going on here is increasing significantly. Thoughts on encapsulating this credential retrieval logic within the ticket service, for maintainability/readability's sake?

Reply:
Let me see what I can figure out.  That's essentially what it was before, and only the username was returned.  When this transitioned to also providing a password I re-arranged things so that it would be a little more straight-forward to retrieve both the username and the password from the AttributePrincipal object without having to try to pass a HashMap or something like that.  If you have a direction in mind let me know...otherwise I'll just take a stab at something and see what I can figure out.

Reply:
Okay, I've moved a bunch of the decryption logic over to the TicketValidationService class and am returning a CredentialsObject back to the AuthenticationProviderService, instead.  How does that look?